ProcessOptimizer/space/constraints.py

```diff
@@ -121,6 +124,100 @@ def rvs(self, n_samples=1, random_state=None):
+        origin = np.zeros(d)
+        delta = np.zeros(d)        
+        for i, dim in enumerate(self.sum_equals[0].dimensions):
+            origin[i] = self.space.bounds[dim][0]
+            delta[i] = self.space.bounds[dim][1] - self.space.bounds[dim][0]
+        A = np.zeros((d,d))
+        B = np.zeros(d)
+        # Row representing the sum constraint
+        A[0,:] = 1
+        B[0] = self.sum_equals[0].value
+        # Rows that define the linear equation for the diagonal along the 
+        # constrained dimensions
+        for i in range(1,d):
+            A[i, i-1] = -delta[i]/delta[0]
+            A[i, i] = 1
+        # Identify the point that lies on the constraint plane and on the diagonal
+        point = np.linalg.solve(A, B)
+        # Use the fact that the vector [1, 1, ...] (a 1 for each constrained 
+        # dimension) is normal to the plane defined by A + B + ... to build
+        # basis-vectors inside the plane, using the null_space function
+        N = matrix(np.ones(d))
+        ns = linalg.null_space(N)
+        # We only need to simulate points up to a distance of half the diagonal
+        # from the origin to A_max, B_max, etc.
+        sim_distance = np.sqrt(np.sum(delta**2)) / 2
+        
+        # Build a list of samples
+        samples = []
+        while len(samples) < n_samples:
+            # Simulate lengths of each null_space vector to add to our point
+            vector_components = ((np.random.rand(d-1, 1) - 0.5)*sim_distance*2)
+            # Generate the samples by adding these different lengths of the null 
+            # space basis vectors to the point on the plane we identified earlier
+            sample_candidate = point[:, None] + ns @ vector_components
+            # Generate the correct shape 
+            sample_candidate = sample_candidate.T[0]
+            # Check that the candidate is inside the original parameter space
+            inspace = []
+            for i in range(d):
+                inspace.append(
+                    np.all(
+                        [(sample_candidate[i] >= self.space.bounds[i][0]),
+                         (sample_candidate[i] <= self.space.bounds[i][1]),],
+                        axis=0,
+                    )
+                )
+            # Only accept the candidate if it is in our space
+            if all(inspace):
+                samples.append(sample_candidate)
+            
+        # Convert our list of samples to an array
+        samples = np.array(samples)
+        
+        # Create settings for the dimensions that are not part of the constraint
+        if d < self.space.n_dims:
+            for i in range(self.space.n_dims):
+                # Only generate settings for the dimensions that are not part 
+                # of the constraint
+                if i not in self.sum_equals[0].dimensions:
+                    dim = self.space.dimensions[i]
+                    column = dim.rvs(n_samples=n_samples, random_state=rng)
```

Comment:
```suggestion
                    column = self.space.rvs(n_samples=n_samples, random_state=rng)
```

The way, we don't interact directly with the dimensions, which makes for cleaner interfaces.

Reply:
The change suggested here is a bit more involved to implement, since the settings for the remaining dimensions have to be inserted in the right place in the full list of samples. As a result, the list comprehension implementation that uses self.space.rvs looks like this (picture because GitHub's code blocks don't respect formatting...):
![image](https://user-images.githubusercontent.com/64902587/226834679-11c6767f-3b7f-463a-b566-76aa5490ca18.png)
While the present code looks like this:
![image](https://user-images.githubusercontent.com/64902587/226834881-c53334b6-7ef5-4284-93bd-d032229c1c3d.png)

While the above is "longer", the more I stare at it I think it is easier to read. Functionally they are the same. I'd like @sqbl to be the impartial judge of readability here, and can just keep whichever he judges most understandable.

Reply:
I agree, either works, so let's use readability, and let @sqbl be the judge of that.

Reply:
I would chose the upper one (the one with "remaining_dimensions").